schema.zmodel

```diff
@@ -0,0 +1,116 @@
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+generator client {
+  provider = "prisma-client-js"
+}
+
+// Here's the simplified model
+//   - Permission is changed to enum so that using it doesn't require join models
+
+// Let's use this project to collaborate
+
+enum Permission {
```

Comment:
I feel permission doesn't need to have its own model, so I changed it to enum and removed the joint tables.

Reply:
But with this I lose the ability to make a custom permission, like a user can only create and edit teams, but cannot delete it, no?

Reply:
Oh, yes. We can change the `postPermission` field of `User` to an array, you can then set it as [CREATE, READ]. 

```prisma
model User {
  id String @id @default(cuid())
  groups GroupUser[]
  posts PostUser[]
  postPermissions Permission[]
}
```

One missing thing is that today in access policy rules there's no way to do array check, but I think that can be easily added, something like:

```@@allow(contains(postPermissions, CREATE))```

That'll work for your needs, right?